src/main/java/uk/gov/ch/developer/docs/controller/developer/UserCallbackController.java

```diff
@@ -41,22 +52,90 @@ public String getCallback(@RequestParam("state") String state,
             // Nonce values
         }
 
-        WebClient webClient = WebClient.create();
+        final Session chSession = (Session) httpServletRequest
+                .getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY);
 
-        webClient.post().uri(URI.create(identityProvider.getTokenUrl()))
-                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
-                .accept(MediaType.APPLICATION_JSON)
-                .body(BodyInserters.fromPublisher(Mono.just(getPostRequestBody(code)),
-                        String.class))
-                .retrieve().bodyToMono(OAuthToken.class)
-                .subscribe(s -> LOGGER.trace("Access Token: " + s.getAccessToken()));// TODO remove
-                                                                                     // this log
-                                                                                     // line
+        Map<String, Object> signInData = new HashMap<>();
+
+        Mono<OAuthToken> oauthTokenResponse = getOAuthToken(code, webClient);
```

Comment:
For this, use `.block()` with a timeout inside `getOAuthToken(...)` So that you have a POJO `OAuthToken` rather than a delayed instance.

Reply:
How long would you suggest the timeout be?